### PR TITLE
Fix this._watchDog can be undefined

### PR DESCRIPTION
**Pull request by QuentiumYT**
Having the issue sometimes since I installed it, I found that when deactivating and reactivating cause the extension to crash with `this._watchDog is undefined`. Just check if this._watchDog is set before accessing its values

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ export default class DashToDockExtension extends Extension.Extension {
         if (!this._isEnabled || this._statusNotifierWatcher)
             return;
 
-        if (this._watchDog.nameAcquired && this._watchDog.nameOnBus)
+        if (this._watchDog && this._watchDog.nameAcquired && this._watchDog.nameOnBus)
```
Comment (3v1n0) [inline extension.js:83]:
```suggestion
        if (this?._watchDog.nameAcquired && this._watchDog.nameOnBus)
```

Comment (QuentiumYT) [reply extension.js:83]: Shouldn't it be `this._watchDog?.nameAcquired`? And also do the same check on nameOnBus like
```js
if (this._watchDog?.nameAcquired && this._watchDog?.nameOnBus)
```

Comment (3v1n0) [reply extension.js:83]: Sorry, yeah I meant to write `this._watchDog?.nameAcquired`... 

> And also do the same check on nameOnBus

That's not needed because if the 1st one is true, then when checking for the 2nd value we're already sure that's defined.

Comment (QuentiumYT) [reply extension.js:83]: Alright, yeah I got it, it was just for consistency :) will force push tomorrow!

File: extension.js [hunk 2]
```diff
@@ -19,8 +19,8 @@ import * as Extension from 'resource:///org/gnome/shell/extensions/extension.js'
 import * as StatusNotifierWatcher from './statusNotifierWatcher.js';
 import * as Interfaces from './interfaces.js';
 import * as TrayIconsManager from './trayIconsManager.js';
+import * as SettingsManager from './settingsManager.js';
 import * as Util from './util.js';
-import {SettingsManager} from './settingsManager.js';
```
Comment (3v1n0) [inline extension.js:23]: This change is unrelated though, also it may not work as otherwise `SettingsManager.SettingsManager` should be used instead.

File: extension.js [hunk 3]
```diff
@@ -80,7 +80,7 @@ export default class DashToDockExtension extends Extension.Extension {
         if (!this._isEnabled || this._statusNotifierWatcher)
             return;
 
-        if (this._watchDog.nameAcquired && this._watchDog.nameOnBus)
+        if (this._watchDog?.nameAcquired && this._watchDog.nameOnBus)
```
Comment (3v1n0) [inline extension.js:83]: Mh, thinking again... I feel we should instead doing something like:

```suggestion
        if (!this._watchDog)
            return;

        if (this._watchDog.nameAcquired && this._watchDog.nameOnBus)
```

As if the watchdog is not set then we should really not pass it later to the status notifier watcher...

**Comment by 3v1n0** [comment]
Do you have a JS trace in the case of `this._watchDog` being undefined?

File: extension.js (hunk 2 above)
Comment (QuentiumYT) [reply extension.js:23]: Ooops was a leftover for uniformization, will remove it

File: extension.js (hunk 3 above)
Comment (QuentiumYT) [reply extension.js:83]: That is way cleaner yup, will do that soon

**Comment by QuentiumYT** [comment]
> Do you have a JS trace in the case of `this._watchDog` being undefined?

Actually no, but I can rebuilt it from main and send it. As I remember, it was triggered on disable / reenable and a few other cases. But there wasn't a lot of details in the error

**Comment by QuentiumYT** [comment]
Done, do you want me to still send the JS error log?

**Comment by 3v1n0** [comment]
I'd be curious, to see if we can figure out also the root reason for this.

**Comment by QuentiumYT** [comment]
I wasn't able to reproduce, very strange. I might have a clue, I have 4 sessions for the desktop environment:
- Ubuntu on xorg
- Ubuntu on wayland
- Gnome on xorg
- Gnome on wayland

And one day it broke almost all extensions (like unable to enable them), this error was thrown when enabling the extension.

I should mention as well that I build glib from source which might lead to some issues. Because I know gio is a bit broken on my laptop (btw if you have enough knowledge to exchange about that ^^) but seems to work fine after I changed a whole number of things

PS: yeah I know, I like to play with fire building a lot of things from source haha, but I enjoy doing that :)
